nixos/modules/services/misc/ollama.nix

```diff
@@ -11,22 +11,47 @@ in
   imports = [
     (lib.mkRemovedOptionModule [ "services" "ollama" "listenAddress" ]
       "Use `services.ollama.host` and `services.ollama.port` instead.")
+    (lib.mkRemovedOptionModule [ "services" "ollama" "sandbox" ]
+      "Set `services.ollama.user` and `services.ollama.group` instead.")
   ];
 
   options = {
     services.ollama = {
       enable = lib.mkEnableOption "ollama server for local large language models";
       package = lib.mkPackageOption pkgs "ollama" { };
+
+      user = lib.mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "ollama";
+        description = ''
+          User account under which to run ollama. Defaults to [`DynamicUser`](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=)
+          when set to `null`.
+        '';
+      };
+
+      group = lib.mkOption {
+        type = with types; nullOr str;
+        default = cfg.user;
+        defaultText = literalExpression "services.ollama.user";
+        example = "ollama";
+        description = ''
+          Group under which to run ollama. Only used when `services.ollama.user` is set.
+        '';
+      };
```

Comment:
Is it necessarily a good idea to make the users specify the user and group? Shouldn't they be defined separately, maybe still using the `sandbox` option to specify when to use them?

Something like
``` nix
users = lib.mkIf (cfg.sandbox == false) {
  users.ollama.isSystemUser = true;
  users.ollama.group = "ollama";
  groups.ollama = {};
};
systemd.services.ollama = {
  # omitted
  serviceConfig = (if cfg.sandbox then {
    DynamicUser = true;
  } else {
    User = "ollama";
    Group = "ollama";
  });
  # omitted
};
```

Reply:
https://search.nixos.org/options?channel=24.05&from=0&size=50&sort=relevance&type=packages&query=.sandbox

That's frankly not a good nor common name for that kind of thing.

Reply:
The name can be changed (maybe just to `dynamicUser`), but I feel like having to manually set a name for the service's user is a worse user experience than just setting a boolean option, and unnecessarily complex for the end goal (enabling/disabling dynamic user and filesystem sandboxing).

Maybe I'm just missing something though; if you can, please explain the benefits of manually setting the user and group names.

Reply:
The user/group options are just very common. I would like to stick with the expectations people have picked up frorm other modules.

Reply:
I've never seen it before. But even if it is a well established convention, conventions should only be kept when there is a benefit to doing so, which I don't see here. Is there some circumstance where changing the name of the service's user or group is useful or necessary? If not, then what benefit could there be in requiring the user to specify the name, instead of simply toggling the features that are necessary for their workflow?

Reply:
I think in most cases where this is not desirable, DynamicUser is just left off.

Reply:
> Is there some circumstance where changing the name of the service's user or group is useful or necessary?

It’s useful if you need the service to run as an existing user that also has access to other things the service will integrate with.

I don’t know if that’s relevant in this case, but e.g. perhaps you want to have your web app in the same group as the service so that it can access generated files. This comes up often with ACME certificate generation, for instance.

I agree that this pattern is very common and reasonable to use when possible. If there’s no reason to ever twiddle these settings, then `DynamicUser` should simply be unconditional instead.

Reply:
The reason to change the models dir is that models might be larger than your mountpoint below `/var/lib/ollama`.

Reply:
@emilazy See issue  #304531 and  pr #305076 for motivation and context for why I made it possible to disable `DynamicUser` and exposed `ReadWritePaths` as `writablePaths`.